src/main/java/com/takipi/oss/storage/resources/StatusStorageResource.java

```diff
@@ -0,0 +1,171 @@
+		StatusUtil.appendInfoMessage(sb, hitsSizeName + ": ", StatusUtil.bytesToKbString(mappedData.get(hitsSizeName)));
+		StatusUtil.appendInfoMessage(sb, hitsCountName + ": ", mappedData.get(hitsCountName));
+		StatusUtil.appendInfoMessage(
+				sb, namersSizeName + ": ", StatusUtil.bytesToKbString(mappedData.get(namersSizeName)));
+		StatusUtil.appendInfoMessage(sb, namersCountName + ": ", mappedData.get(namersCountName));
+		StatusUtil.appendInfoMessage(
+				sb, sourceCodesSizeName + ": ", StatusUtil.bytesToKbString(mappedData.get(sourceCodesSizeName)));
+		StatusUtil.appendInfoMessage(sb, sourceCodesCountName + ": ", mappedData.get(sourceCodesCountName));
+		StatusUtil.appendInfoMessage(sb, "Total free space left: ", StatusUtil.bytesToMbString(directory.getFreeSpace()));
+	}
+	
+	private HashMap<String, Long> traverseTreeForData(File directory) {
+		HashMap<String, Long> map = initializeMapForData();
+		
+		if (directory.isDirectory()) {
+			traverseTreeForData(directory, map);
+		}
+		
+		return map;
+	}
+	
+	// Traverse into the directory and it's subdirectories,
+	// avoiding any hidden or visible files until finding
+	// the suitable directory (hits/namers/source-code).
+	private void traverseTreeForData(File directory, HashMap<String, Long> map) {
+		for(File file : directory.listFiles()) {
+			if (file.isHidden() || !file.isDirectory()) {
+				continue;
+			}
+			
+			directoryHandler(map, file);
+		}
+	}
+	
+	// Direct hits/namers/source-code directory to it's handler
+	// or keep traversing.
+	private void directoryHandler(HashMap<String, Long> map, File directory) {
+		switch (directory.getName()) {
+			case hitsDirectoryName : {
+				handleSpecialDirectory(directory, hitsSizeName, hitsCountName, map);
+				break;
+			} 
+			case namersDirectoryName : {
+				handleSpecialDirectory(directory, namersSizeName, namersCountName, map);
+				break;
+			} 
+			case sourceCodeDirectoryName : {
+				handleSpecialDirectory(directory, sourceCodesSizeName, sourceCodesCountName, map);
+				break;
+			} 
+			default : {
+				traverseTreeForData(directory, map);
+			}
+		}
+	}
+	
+	// Extract data of visible files.
+	private void handleSpecialDirectory(File directory, String sizeName, String countName, HashMap<String, Long> map) {
+		for(File file : directory.listFiles()) {
+			if (file.isHidden()) {
+				continue;
+			}
+			
+			if (file.isDirectory()) {
+				handleSpecialDirectory(file, sizeName, countName, map);
+			} else {
+				long sizeValue = map.get(sizeName);
+				long countValue = map.get(countName);
+				
+				map.put(sizeName, sizeValue + safeFileSize(file));
+				map.put(countName, countValue + 1);
+			}
+		}
+	}
+	
+	private long safeFileSize(File file) {
+		try {
+			return FileUtils.sizeOf(file);
+		} catch (Exception e) {
+			return 0;
+		}
+	}
+	
+	private HashMap<String, Long> initializeMapForData() {
+		HashMap<String, Long> result = new HashMap<>();
```

Comment:
Aside from "new HashMap", all other places can be "Map", no need to force HashMap on them